hs_core/hydroshare/utils.py

```diff
@@ -249,20 +258,9 @@ def get_file_from_irods(res_file):
     :return: location of the copied file
     """
     res = res_file.resource
-    if res_file.fed_resource_file or res_file.fed_resource_file_name_or_path:
-        istorage = IrodsStorage('federated')
-    else:
-        istorage = IrodsStorage()
-    if res_file.resource_file:
-        res_file_path = res_file.resource_file.name
-        file_name = os.path.basename(res_file.resource_file.name)
-    elif res_file.fed_resource_file:
-        res_file_path = res_file.fed_resource_file.name
-        file_name = os.path.basename(res_file.fed_resource_file.name)
-    else:
-        res_file_path = os.path.join(res.resource_federation_path, res.short_id,
-                                     res_file.fed_resource_file_name_or_path)
-        file_name = os.path.basename(res_file.fed_resource_file_name_or_path)
+    istorage = res.get_irods_storage()
+    res_file_path = res_file.storage_path
+    file_name = os.path.basename(res_file_path)
```

Comment:
@alvacouch @pkdash OK. I take my comments back. I think it is fine at least in production. You can disregard my comment above. But I will not delete my comment for reference just in case.